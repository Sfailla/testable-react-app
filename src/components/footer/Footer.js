import React from 'react';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles(theme => ({
	footer: {
		width: '100%',
		height: '60px',
		color: theme.palette.common.white,
		backgroundColor: theme.palette.tertiary.main,
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',

		position: 'absolute',
		bottom: '0',
		left: '0'
	}
}));

function Footer() {
	const classes = useStyles();
	return (
		<div className={classes.footer}>
			<p>created by steven failla &copy; 2020</p>
		</div>
	);
}

export default Footer;
