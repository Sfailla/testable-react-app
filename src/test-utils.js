import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'react-jss';
import { theme } from './styles/theme';

const AllMyProviders = ({ children }) => {
	return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const MyCustomRender = (component, options) =>
	render(component, { wrapper: AllMyProviders, ...options });

// re-export everything from the `react-testing-library`
export * from '@testing-library/react';

// export our custom render method
export { MyCustomRender as render };
