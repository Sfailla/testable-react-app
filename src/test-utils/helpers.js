import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'react-jss';
import { theme } from '../styles/theme';
import { StaticRouter } from 'react-router-dom';

function shallowWithTheme(child) {
	return shallow(child, {
		wrappingComponent: ({ children }) => (
			<ThemeProvider theme={theme}>{children}</ThemeProvider>
		)
	});
}

function mountWithTheme(child) {
	return mount(child, {
		wrappingComponent: ({ children }) => (
			<ThemeProvider theme={theme}>{children}</ThemeProvider>
		)
	});
}

function wrapWithRouter({ children }) {
	return <StaticRouter>{children}</StaticRouter>;
}

module.exports = {
	shallowWithTheme,
	mountWithTheme,
	wrapWithRouter
};
