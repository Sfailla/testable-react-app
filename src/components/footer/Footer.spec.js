import React from 'react';
import { render } from '../../test-utils';
import { mountWithTheme } from '../../test-utils/helpers.js';
import toJson from 'enzyme-to-json';

import Footer from './Footer';

let wrapper;

beforeEach(() => {
	wrapper = mountWithTheme(<Footer />);
});

describe('<Footer /> component unit tests', () => {
	it('should render 1 <Footer /> component', () => {
		expect(wrapper.find(Footer)).toHaveLength(1);
	});

	it('should take snapshot that matches component', () => {
		let tree = toJson(wrapper);
		expect(tree).toMatchSnapshot();
	});

	it('should have the same text value as component', () => {
		const text = `created by steven failla ${'&copy;'} 2020`;
		// for decoding html entity.  &copy;
		let elem = document.createElement('textarea');
		elem.innerHTML = text;
		let decoded = elem.value;
		expect(wrapper.find('p').text()).toEqual(decoded);
	});
});

describe('<Footer> unit tests', () => {
	it('renders the footer without error', () => {
		render(<Footer />);
	});

	it('should render footer with correct text', () => {
		const { getByText } = render(<Footer />);
		const text = 'created by steven failla &copy; 2020';
		// for decoding html entity.  &copy;
		let elem = document.createElement('textarea');
		elem.innerHTML = text;
		let decoded = elem.value;
		expect(getByText(decoded)).toBeTruthy();
	});

	it('should take a snapshot', () => {
		const { asFragment } = render(<Footer />);
		expect(asFragment(<Footer />)).toMatchSnapshot();
	});
});
