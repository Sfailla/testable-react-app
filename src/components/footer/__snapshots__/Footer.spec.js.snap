// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component unit tests should take snapshot that matches component 1`] = `
<Footer>
  <footer
    className="footer-0-2-1"
  >
    <p>
      created by steven failla Â© 2020
    </p>
  </footer>
</Footer>
`;
